crates/biome_js_analyze/src/lint/nursery/use_iterable_callback_return.rs

```diff
@@ -0,0 +1,378 @@
+use crate::ControlFlowGraph;
+use crate::services::control_flow::JsControlFlowGraph;
+use biome_analyze::{Rule, RuleDiagnostic, RuleSource, context::RuleContext, declare_lint_rule};
+use biome_console::fmt::{Display, Formatter};
+use biome_console::markup;
+use biome_control_flow::builder::ROOT_BLOCK_ID;
+use biome_control_flow::{ExceptionHandlerKind, InstructionKind};
+use biome_js_syntax::{
+    AnyJsFunctionBody, JsArrowFunctionExpression, JsCallArgumentList, JsCallArguments,
+    JsCallExpression, JsFunctionExpression, JsReturnStatement, global_identifier,
+};
+use biome_rowan::{AstNode, AstSeparatedList, NodeOrToken, TextRange};
+use roaring::RoaringBitmap;
+use rustc_hash::FxHashMap;
+use std::io;
+use std::sync::LazyLock;
+
+declare_lint_rule! {
+    /// Enforce consistent return values in iterable callbacks.
+    ///
+    /// This rule ensures that callbacks passed to certain iterable methods either always return a value or never return a value, depending on the method's requirements.
+    ///
+    /// ## Methods and Their Requirements
+    ///
+    /// - Methods like `map`, `filter`, `reduce`, `some`, `every`, etc., require the callback to always return a value.
+    /// - Methods like `forEach` do not require the callback to return a value.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// [].map(() => {
+    ///     // Missing return value
+    /// });
+    /// ```
+    ///
+    /// ```js,expect_diagnostic
+    /// [].forEach(() => {
+    ///     return 1; // Should not return a value
+    /// });
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// [].map(() => {
+    ///     return 1; // Correctly returns a value
+    /// });
+    /// ```
+    ///
+    /// ```js
+    /// [].forEach(() => {
+    ///     // No return value, which is correct
+    /// });
+    /// ```
+    pub UseIterableCallbackReturn {
+        version: "next",
+        name: "useIterableCallbackReturn",
+        language: "js",
+        sources: &[RuleSource::Eslint("array-callback-return")],
+        recommended: false,
```

Comment:
```suggestion
        recommended: true,
```

I think this rule should be recommended (don't worry, at compile time this value gets changed for nursery rules), because I presume we will promote this rule to the `correctness` group. 

Also, I would change the severity to `Severity::Error`. Can you do that? I can't propose the suggestion because you'll need to import `Severity` too

Reply:
Done